Extension/src/LanguageServer/protocolFilter.ts

```diff
@@ -23,6 +46,24 @@ export function createProtocolFilter(me: Client, clients: ClientCollection): Mid
         didOpen: (document, sendMessage) => {
             if (clients.checkOwnership(me, document)) {
                 me.TrackedDocuments.add(document);
+
+                // Loop through registered providers until one is able to service the current document
+                customConfigurationProviders.some((provider: CustomConfigurationProvider): boolean => {
+                    let canProvide: boolean = provider.canProvideConfiguration(document.uri);
+                    if (canProvide) {
+                        runThenableWithTimeout(
```

Comment:
This method should block execution.  You can update `canProvide` based on the result of this function.  When you successfully get a config, you can just fall out of the loop and use the `sendMessage(document);` call at the end of the method instead of including it here.